Edit getting started docs (#375)
telerik/blazor-docs: _contentTemplates/common/get-started.md

#add-nuget-feed

> ### Add the Telerik NuGet Feed to Visual Studio

Author: The article relies heavily on using NuGet as a package source. Why do we include "#step-0---download-the-components" in the same article?

Reviewer: This is extremely complicated. Downloading the components lets the Telerik Control Panel set up the nuget feed for people so they can usually skip parts of the tutorial. It also gives them the local demos and an option for a local feed in case they can't get to our online feed. That download section is also required by management to boost trial downloads. The whole tutorial is far too complex with far too many options and different config setups for it to be straightforward. For example, the trial experience from downloading the web installer is completely different from the setup for a person who has DCC and has had a commercial version for ages - the latter probably has the nuget feed setup already, for one, while the former sometimes does not even know what Telerik is.

> ### Add the Telerik NuGet Feed to Visual Studio

Author: This section addresses the needs of the trial users but not of the users with commercial licenses. Should we add something for the users with license here?

Reviewer: The two tutorial-like articles target trial users for the most part. This is designed so by management (hence the "download the components" section too).

> Telerik UI for Blazor is distributed through our private NuGet feed.

Reviewer: Effectively, it is. For tutorial-like articles it is best to make things clear, there are still many people who wan to reference individual `dll` files. For advanced users, we have other articles that tell them how they can download `nupkg` files on their own. I propose we keep this line in the tutorials.

> Telerik UI for Blazor is distributed through our private NuGet feed.

Author: I see how important NuGet is as a distribution method. This brings again the question of having another article in the Getting Started section that covers the MSI scenario. Covering both in a single article makes the article too complex.

> Telerik UI for Blazor is distributed through our private NuGet feed.

Reviewer: We have the three installation methods (including msi and nuget) in three separate articles under Installation.

> Telerik UI for Blazor is distributed through our private NuGet feed.

Author: Back to the beginning of the discussion - we cannot keep the original line because it introduces a conflict in this article (the article starts with instructions on how to download an installer and then claims that NuGet is the only distribution method). 
To underline the significance of NuGet, I can change the line to something like:
"The recommended distribution method for the Telerik UI for Blazor packages is Telerik's private NuGet feed."

> One of the distribution options for Telerik UI for Blazor is a private NuGet feed.

Author: ```suggestion
The recommended distribution method for the Telerik UI for Blazor packages is the private Telerik NuGet feed.
```

One of the distribution options for Telerik UI for Blazor is a private NuGet feed.

@[template](/_contentTemplates/common/get-started.md#start-trial)

If you prefer to do the process yourself, follow the [Setup the Telerik Private NuGet Feed]({%slug installation/nuget%}) article to set it up manually in case you don't have it already.
If you prefer to configure the NuGet package source manually, follow the steps in the [Setup the Telerik Private NuGet Feed]({%slug installation/nuget%}) article.

Once you have added the Telerik NuGet feed, continue with this tutorial.
Once you have added the Telerik NuGet feed, continue by [enabling the components in the project](#enable-the-components-in-the-project).
#end

#project-creation-part-1
 If you have one, go to the [Add the Telerik Blazor Components to an Existing Project](#step-2---add-the-telerik-blazor-components-to-an-existing-project) section below.

> There are three ways to create the project:

Reviewer: your change is good

  * [Create a Project with Visual Studio](#create-a-project-with-visual-studio).
  * [Create a Project with the CLI](#create-a-project-with-the-cli).

> ### Create a Project with the Telerik VS Extensions

Reviewer: I think that's what the very first sentence does? If it is unclear, we could probably reword it further to start with "To start using the components immediately, we can create and configure a project for you through or vs extensions".

> ### Create a Project with the Telerik VS Extensions

Author: This should be stated more clearly.

* [Create a Project with the CLI](#create-a-project-with-the-cli)
* To create a new project, choose one of the following methods:

* [Create a Project with Visual Studio](#create-a-project-with-visual-studio)
  * [Create a Project with the Telerik VS Extensions](#create-a-project-with-the-telerik-vs-extensions).
  * [Create a Project with Visual Studio](#create-a-project-with-visual-studio).
  * [Create a Project with the CLI](#create-a-project-with-the-cli).

### Create a Project with the Telerik VS Extensions

You can use the [Visual Studio Extensions]({%slug getting-started-vs-integration-overview%}) we provide to [create the project for you]({%slug getting-started-vs-integration-new-project%}), so that you can start using the Telerik components immediately.
You can use the [Telerik Visual Studio Extensions]({%slug getting-started-vs-integration-overview%}) that will [create and automatically configure the project]({%slug getting-started-vs-integration-new-project%}) so that you can start using the components immediately.

> If you use VS Code, you can also use our [VS Code Extension to create a Telerik-enabled project]({%slug getting-started-vs-code-integration-overview%}).

Reviewer: The feedback we had (I am not aware of newer) is that VS Code is used quite rarely with Blazor, which is why I didn't mention it heavily or stress on it. We can keep it as a tip.

>tip If you prefer VS Code, you can use the [VS Code Extension]({%slug getting-started-vs-code-integration-overview%}) to create a Telerik-enabled project.

The rest of this article will explain the manual steps if you want to have a better understanding of the underlying process.
If you choose to create a project with the Telerik VS Extensions, you can jump directly to [Step 3 - Add a Telerik Component to a View](#step-3---add-a-telerik-component-to-a-view). The following sections in this article explain the manual steps to configure the project so that you can better understand the underlying process.

> ### Create a Project with the CLI

Reviewer: I want to keep the section so there is a heading for SEO and for visibility, and I do not see a reason for much more details because CLI users generally know what the CLI is and how to use it.

I want to keep the extensions as the first section because it is the easiest for people. In fact, if they use our extensions, they don't have to follow this article further at all, the extensions will do everything for them. Not sure if we should remove the link to Step 3

> ### Create a Project with the CLI

Author: I agree that keeping the extensions as the first step is a good approach. My suggestion about removing the "Create a Project with the CLI" sections comes from the desire to have a leaner article.

### Create a Project with Visual Studio

The next section shows how to create the Blazor through the Visual Studio interface. If you are not running Visual Studio, you can create the Blazor project from the command prompt - see the `dotnet new` command and the arguments related to Blazor apps: 
To create a project manually, without using the Telerik VS Extensions, follow these steps:

* <a href="https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new#arguments" target="_blank">MSDN: dotnet new arguments</a>
1. Open Visual Studio 2019.

> The rest of this article shows some steps through the Visual Studio interface, as it is the most common IDE used with Blazor. Where such steps exist, there are links and instructions on performing them yourself (e.g., adding a nuget feed through the CLI).

Reviewer: They were added to avoid support tickets and confusion in some customers. I recommend putting them back in.

> The rest of this article shows some steps through the Visual Studio interface, as it is the most common IDE used with Blazor. Where such steps exist, there are links and instructions on performing them yourself (e.g., adding a nuget feed through the CLI).

Author: The need for these clarifications is again related to the complexity of the article. This is one of the signs showing that the article could use some simplification.

1. Create a New Project.

1. Choose **Blazor App** and click **Next**. Then, choose a name and location for the project and click **Create**.

### Create a Project with Visual Studio
    ![Create new ASP.NET Core Web Application](images/create-new-application.png)

To create a project manually, follow these steps:
#end

1. Open Visual Studio 2019

1. Create a New Project
#project-creation-cli
### Create a Project with the CLI

> 1. Choose `Blazor App` and click `Next`. Then, choose a name and location for the project and click `Create`.

Reviewer: It seems they have changed it and it is now "Blazor Server App` and "Blazor WebAssembly App", fixing this will require heavier edits and maybe killing this as a template.

If you are not running Visual Studio, you can create the Blazor project from the command prompt. See the <a href="https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new#arguments" target="_blank">`dotnet new` command and the arguments for Blazor apps</a>.

    ![Create new ASP.NET Core Web Application](images/create-new-application.png)
#end

#start-trial

> If you don't have an active license, [start a UI for Blazor trial](https://www.telerik.com/download-trial-file/v2-b/ui-for-blazor). The file this will let you download and install the components, and will also let you setup our [online NuGet feed]({%slug installation/nuget%}) automatically - make sure to select the "Set up Telerik NuGet package source" checkbox: 

Author: This is repeating the information in step 0 - download the free trial. Do we need it?

Reviewer: Yes, it takes care of customer questions and tickets and avoids confusion.

#get-access
### Enable the Components in the Project

> To have the project use the Telerik UI for Blazor components, follow these steps:

Author: This is the 3rd time we ask the reader to start a free trial and download the components. Do we need it?

Reviewer: Added to reduce confusion and avoid tickets. I recommend putting it back in. Trial uses should not have to wait a few days for us to tell them "read this article to get the feed running".

#download-intro-para-for-get-started
If you are familiar with the Telerik NuGet Feed and Blazor in general, you may want to follow the shorter, more technical article with the same information: [What You Need]({%slug getting-started/what-you-need%}). The current article is designed as a step-by-step tutorial from the basics for new users.
If you are familiar with the Telerik NuGet Feed and Blazor in general, you may want to follow the shorter, more technical getting started article: [What You Need]({%slug getting-started/what-you-need%}). The current article is designed as a step-by-step tutorial for new users and starts from the basics.

## Step 0 - Download the Components

> A pre-requisite is having access to the Telerik UI for Blazor components. The easiest way to get them to your development machine is to use the <a href="https://www.telerik.com/download-trial-file/v2/control-panel" target="_blank">Progress Control Panel</a> or to download the [automated installer]({%slug installation/msi%}) from <a href="https://www.telerik.com/account/product-download?product=BLAZOR" target="_blank">your telerik.com account</a>.

Author: The tutorials should also focus on the trial users as an important group. That's why I suggest adding two sub-sections under "Step 0 - Download the Components" - the first section speaks to trialists and the second section speaks to developers with a license.
